settings-boundary/EventSettings.tsx

```diff
@@ -0,0 +1,238 @@
+import { Placemark } from "TiFShared/domain-models/Placemark"
+import { formatEventDurationPreset } from "TiFShared/domain-models/Settings"
+import React from "react"
+import {
+  StyleProp,
+  StyleSheet,
+  TouchableHighlight,
+  TouchableOpacity,
+  View,
+  ViewStyle
+} from "react-native"
+import { SettingsNamedToggleView } from "./components/NamedToggle"
+import { SettingsNavigationLinkView } from "./components/NavigationLink"
+import { SettingsScrollView } from "./components/ScrollView"
+import {
+  SettingsCardSectionView,
+  SettingsSectionView
+} from "./components/Section"
+
+export type SettingDurationCardProps = {
+  style?: StyleProp<ViewStyle>
+  durationInSeconds: number
+}
+
+export const SettingsDurationCard = ({
+  style,
+  durationInSeconds
+}: SettingDurationCardProps) => {
+  return (
+    <View style={styles.container}>
+      <Headline>{formatEventDurationPreset(durationInSeconds)}</Headline>
+      <TouchableOpacity style={styles.closeButton}>
+        <Ionicon size={16} color={"white"} name={"close"} />
+      </TouchableOpacity>
+    </View>
+  )
+}
+
+export const AddDurationCard = () => {
+  return (
+    <TouchableHighlight style={styles.addButtonContainer}>
+      <Ionicon size={36} color={AppStyles.colorOpacity35} name={"add"} />
+    </TouchableHighlight>
+  )
+}
+
+export const DurationSectionView = () => {
+  const { settings } = useUserSettings(settingsSelector("eventPresetDurations"))
+  const sortedDurations = settings.eventPresetDurations.sort((a, b) => a - b)
+  return (
+    <SettingsSectionView title="Duration Presets">
+      {sortedDurations.length < 3 ? (
+        <View style={styles.presetRowsGridContainer}>
+          <View style={styles.durationPresetRow}>
+            {createDurationCards(0, sortedDurations, 3)}
+            {settings.eventPresetDurations.length < 3 ? (
+              <AddDurationCard />
+            ) : undefined}
+            {settings.eventPresetDurations.length < 2 ? (
+              <View style={[styles.container, { opacity: 0 }]} />
+            ) : undefined}
+            {settings.eventPresetDurations.length < 1 ? (
+              <View style={[styles.container, { opacity: 0 }]} />
+            ) : undefined}
+          </View>
+        </View>
+      ) : (
+        <View style={styles.presetRowsGridContainer}>
+          <View style={styles.durationPresetRow}>
+            {createDurationCards(0, sortedDurations, 3)}
+          </View>
+          <View style={styles.durationPresetRow}>
+            {createDurationCards(3, sortedDurations)}
+            {settings.eventPresetDurations.length < 6 ? (
+              <AddDurationCard />
+            ) : undefined}
+            {settings.eventPresetDurations.length < 5 ? (
+              <View style={[styles.container, { opacity: 0 }]} />
+            ) : undefined}
+            {settings.eventPresetDurations.length < 4 ? (
+              <View style={[styles.container, { opacity: 0 }]} />
+            ) : undefined}
```

Comment:
Be sure to have this be more general (rows and columns / grid component with items)